### ISSUE TITLE
jlink: Hash of java.rmi differs to expected hash recorded in java.base

### ISSUE THREAD

**Issue by gorankarlic** [oracle/graal#3965]
Error when using `jlink` with GraalVM CE 21.3.0 on Windows x64:

```bash
> bin\jlink --output jre-11.0.13 --add-modules java.base,java.rmi
Error: Hash of java.rmi (f671e42903a0adce0642b577802ce36274988cf9a88dc3b2115be3d6cd7e5677) differs to expected hash (abb5652878595f7bbe89256539153df4c639d2446b5baa127cf71ef8d4d18d8b) recorded in java.base
```

```bash
> bin\java --version
openjdk 11.0.13 2021-10-19
OpenJDK Runtime Environment GraalVM CE 21.3.0 (build 11.0.13+7-jvmci-21.3-b05)
OpenJDK 64-Bit Server VM GraalVM CE 21.3.0 (build 11.0.13+7-jvmci-21.3-b05, mixed mode, sharing)
```

_Originally posted by @gorankarlic in https://github.com/oracle/graal/issues/3090#issuecomment-955668637_

**Comment by rodrigar-mx** [comment]
Hi @gorankarlic. Thanks for reporting this issue. I will take a look into it and report the bug if required.

**Comment by rodrigar-mx** [comment]
The same error is produced in macOS with GraalVM CE 21.3.0 JDK 11 version, however, in macOS with GraalVM CE 21.3.0 JDK 17 or Linux AMD 64 with GraalVM CE 21.3.0 JDK 11 or with GraalVM CE 21.3.0 JDK 17 no error is generated.

The bug has now been reported internally. An answer will be provided accordingly.

**Comment by blackjyn** [comment]
I got similar error:
```
Error: Hash of jdk.jpackage (7b29cf91a3423c4473757d30e37d6d6f8cf568a70f6f1116824
31396df7ab38f) differs to expected hash (c3b4618bbb094fe2cb009cc29e0f0e182c8fe89ff3001bf7cbe4f0612c394706) recorded in java.base 
java.lang.module.FindException: Hash of jdk.jpackage (7b29cf91a3423c4473757d30e37d6d6f8cf568a70f6f111682431396df7ab38f) 
differs to expected hash (c3b4618bbb094fe2cb009cc29e0f0e182c8fe89ff3001bf7cbe4f0612c394706) recorded in java.base
```

**Comment by gilles-duboscq** [comment]
This should be fixed in the upcoming January releases (22.0.0, 21.3.1)
